drivers: ieee802154: get eui64 MAC address function updated

xavraz opened the pull request:
The mcxw_get_eui64() function allows to provide a persistent MAC address if it does not exist

axelnxp commented on drivers/ieee802154/ieee802154_mcxw.c:
```diff
@@ -81,6 +87,11 @@ static uint8_t ot_phy_ctx = (uint8_t)(-1);
 
 static struct mcxw_context mcxw_ctx;
 
+static int thread_mac_settings_set(const char *name, size_t len,
```
this is a 802.15.4 driver, not specific to thread itself, so I wouldn't use "thread" keyword here (same thing for the macros and other variables).

jukkar reviewed (commented):
Shouldn't the link address value come from DT or from device rom etc?
I am not sure this is a good idea to have settings db used by the device driver as the settings api feels it belongs to a higher level, like to application or ieee802154 subsystem.

George-Stefan commented on drivers/ieee802154/ieee802154_mcxw.c:
```diff
@@ -95,14 +105,113 @@ WEAK void app_disallow_device_to_slepp(void)
+	{
+		LOG_INF("Generating new MAC address");
+
+		/* Set NXP OUI (00:04:9F) for the first 3 bytes
```
comment is not correct

xavraz commented on drivers/ieee802154/ieee802154_mcxw.c:
```diff
@@ -95,14 +105,113 @@ WEAK void app_disallow_device_to_slepp(void)
+		g_eui64[0] = 0x00;
+		g_eui64[1] = 0x60;
+		g_eui64[2] = 0x37;
+		g_eui64[0] = (g_eui64[0] & ~0x01) | 0x02;
```
Hi George,
When the flash has been erased : permit to generate a new eui64.

JarEmb commented:
> address value come from DT or from device rom etc?

@jukkar You mean define a DT storage partition in the Flash to save eui64  and use NVS driver instead of settings to access it?

xavraz committed:
boards: nxp: mcxw72: update flash code/hw_params/storage allocation

This commit add 1K for the hardware parameters (at the end of the
flash) and reduces the storage to 15k. It is used, first, to store
the MAC address.

Signed-off-by: Xavier Razavet <xavier.razavet@nxp.com>

xavraz committed:
boards: nxp: frdm_mcxw71: update flash code/hw_params/storage allocation

This commit add 1K for the hardware parameters (at the end of the flash)
and reduces the storage to 111k. It is used, first, to store the MAC
address.

Signed-off-by: Xavier Razavet <xavier.razavet@nxp.com>

xavraz committed:
samples: openthread: shell: prj-ot-host.conf file updated

Settings configuration is removed in the
zephyr/samples/net/openthread/shell/prj-ot-host.conf file.
No more used.

Signed-off-by: Xavier Razavet <xavier.razavet@nxp.com>

xavraz committed:
drivers: ieee802154: mcxw_get_eui64() updated to manage MAC address

The mcxw_get_eui64() function allows to provide a persistent MAC
address if it does not exist

Signed-off-by: Xavier Razavet <xavier.razavet@nxp.com>

axelnxp commented on boards/nxp/frdm_mcxw72/frdm_mcxw72_mcxw727c_cpu0.dts:
```diff
@@ -135,7 +135,12 @@
 
 		storage_partition: partition@1FC000 {
-			reg = <0x1FC000 DT_SIZE_K(16)>;
+			reg = <0x1FC000 DT_SIZE_K(15)>;
```
NVS needs at least 2 sectors, and on W72/W71 1 sector is 8k (0x2000).

xavraz replied:
Hi Axel,
It is corrected, thanks.

axelnxp commented on boards/nxp/mcxw72_evk/mcxw72_evk_mcxw727c_cpu0.dts:
```diff
@@ -135,7 +135,12 @@
+
+		hw_params_partition: partition@1FFC00 {
+			label = "hw-parameters";
+			reg = <0x1FB000 DT_SIZE_K(1)>;
```
same

axelnxp reviewed (changes requested):
Partition mapping is incorrect:
* NVS needs 2 sectors, so 16k
* Your new partition should be sector aligned, because the erase is done by sector
* Some errors in DTS regarding addresses, and sizes (1k != 0x1000)

axelnxp commented on boards/nxp/frdm_mcxw71/frdm_mcxw71.dts:
```diff
@@ -189,7 +189,12 @@
+
+		hw_params_partition: partition@FE000 {
+			label = "hw-parameters";
+			reg = <0x1FB000 DT_SIZE_K(8)>;
```
Should be:
```suggestion
			reg = <0xFE000 DT_SIZE_K(8)>;
```

axelnxp commented on boards/nxp/frdm_mcxw72/frdm_mcxw72_mcxw727c_cpu0.dts:
```diff
@@ -131,12 +131,17 @@
+		hw_params_partition: partition@1FC000 {
+			label = "hw-parameters";
+			reg = <0x1FB000 DT_SIZE_K(8)>;
+		};
```
Place it at the last sector of the flash
```suggestion
		hw_params_partition: partition@1FE000 {
			label = "hw-parameters";
			reg = <0x1FE000 DT_SIZE_K(8)>;
		};
```

axelnxp commented on boards/nxp/frdm_mcxw72/frdm_mcxw72_mcxw727c_cpu0.dts:
```diff
@@ -131,12 +131,17 @@
-		storage_partition: partition@1FC000 {
+		storage_partition: partition@1FA000 {
 			reg = <0x1FC000 DT_SIZE_K(16)>;
 		};
```
Make sure the start address matches.
```suggestion
		storage_partition: partition@1FA000 {
			reg = <0x1FA000 DT_SIZE_K(16)>;
		};
```

axelnxp reviewed (changes requested):
Still some issues in the DTS mapping

xavraz commented on samples/net/openthread/shell/prj-ot-host.conf:
```diff
@@ -28,9 +25,6 @@ CONFIG_SHELL_DEFAULT_TERMINAL_WIDTH=1024
 CONFIG_FLASH=y
 CONFIG_FLASH_MAP=y
-CONFIG_SETTINGS=y
-CONFIG_SETTINGS_NVS=y
```
In .config file, settings flags are set as follows :
...
CONFIG_SETTINGS=y
...
CONFIG_SETTINGS_LOG_LEVEL_DEFAULT=y
CONFIG_SETTINGS_LOG_LEVEL=3
...
CONFIG_SETTINGS_DYNAMIC_HANDLERS=y
CONFIG_SETTINGS_NVS=y
...
